cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/functional/BasicLatticesRedesign.kt

```diff
@@ -138,16 +138,36 @@ interface Lattice<T : Lattice.Element> {
         transformation: (Lattice<T>, EvaluationOrder, T) -> T,
     ): T {
         val globalState = IdentityHashMap<EvaluationOrder, T>()
-        val finalState: T = this.bottom
+        var finalState: T = this.bottom
         for (startEdge in startEdges) {
             globalState[startEdge] = startState
         }
-        val edgesList = mutableListOf<EvaluationOrder>()
-        startEdges.forEach { edgesList.add(it) }
-
-        while (edgesList.isNotEmpty()) {
-            val nextEdge = edgesList.first()
-            edgesList.removeFirst()
+        val currentBBEdgesList = mutableListOf<EvaluationOrder>()
+        val potentialNextBBEdgesList = mutableListOf<EvaluationOrder>()
```

Comment:
I think the naming here could be improved. I would rename `potentialNextBBEdgesList` to `nextBranchEdgeList` or `nextBBEdgesList`. Additionally I could see `mergePointsEdgesList` to be better named `edgesAfterMergePoint`.